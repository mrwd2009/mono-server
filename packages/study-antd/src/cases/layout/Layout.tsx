import { FC } from 'react';
import Layout from '../../components/layout';
import './index.less';

export interface Props {
  type: string;
  theme: 'dark-theme' | 'light-theme';
}

const LayoutCase: FC<Props> = ({ type, theme = 'light-theme' }) => {
  return (
    <div className={theme}>
      <Layout>
        <Layout.Sider>Sider</Layout.Sider>
        <Layout.Content>Content</Layout.Content>
      </Layout>
    </div>
  );
};

export default LayoutCase;