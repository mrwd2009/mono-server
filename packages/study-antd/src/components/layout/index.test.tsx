import renderer from 'react-test-renderer';
import Layout from './index';


test('layout with full elements', () => {
  const component = renderer.create((
    <Layout>
      <Layout.Sider>Sider</Layout.Sider>
      <Layout.Content>
        <Layout.Header>Header</Layout.Header>
        <Layout.Footer>Footer</Layout.Footer>
      </Layout.Content>
    </Layout>
  ));
  expect(component.toJSON()).toMatchSnapshot();
});