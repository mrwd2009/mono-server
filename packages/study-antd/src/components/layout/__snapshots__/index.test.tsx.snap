// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout with full elements 1`] = `
<section
  className="ant-layout"
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark"
    style={
      Object {
        "flex": "0 0 200px",
        "maxWidth": "200px",
        "minWidth": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      Sider
    </div>
  </aside>
  <main
    className="ant-layout-content"
  >
    <header
      className="ant-layout-header"
    >
      Header
    </header>
    <footer
      className="ant-layout-footer"
    >
      Footer
    </footer>
  </main>
</section>
`;
